lib/kernel/src/inet_db.erl

```diff
@@ -1621,46 +1618,41 @@ do_add_rr(RR, Db, State) ->
     TM = times(),
     case alloc_entry(Db, CacheDb, TM) of
 	true ->
-	    cache_rr(Db, CacheDb, RR#dns_rr{tm = TM, cnt = TM});
+	    cache_rr(CacheDb, RR#dns_rr{tm = TM, cnt = TM});
 	_ ->
 	    false
     end.
 
-cache_rr(_Db, Cache, RR) ->
-    %% delete possible old entry
-    ets:match_delete(Cache, RR#dns_rr{cnt = '_', tm = '_', ttl = '_',
-				      bm = '_', func = '_'}),
-    ets:insert(Cache, RR).
+cache_rr(Cache, RR) ->
+    ets:insert(Cache, {cache_key(RR), RR}).
 
 times() ->
     erlang:monotonic_time(second).
     %% erlang:convert_time_unit(erlang:monotonic_time() - erlang:system_info(start_time),
     %%     		     native, second).
 
-%% lookup and remove old entries
-
-do_lookup_rr(Domain, Class, Type) ->
-    match_rr(#dns_rr{domain = tolower(Domain), class = Class,type = Type,
-		     cnt = '_', tm = '_', ttl = '_',
-		     bm = '_', func = '_', data = '_'}).
-
-match_rr(RR) ->
-    filter_rr(ets:match_object(inet_cache, RR), times()).
-
-
-%% filter old resource records and update access count
+%% match and remove old entries
+
+match_rr({_, _, _} = Key) ->
+    Time = times(),
+    case ets:lookup(inet_cache, Key) of
+	[{_,RR}] when RR#dns_rr.ttl =:= 0 -> %% at least once
+	    ets:delete(inet_cache, Key),
+	    [RR];
+	[{_,RR}] when RR#dns_rr.tm + RR#dns_rr.ttl < Time ->
+	    ets:delete(inet_cache, Key),
+	    [];
```

Comment:
These lines are not covered in our daily tests by running all test suites in Kernel.

Reply:
Was this line covered in the previous implementation? I am asking just to know if it is something in my implementation or something that carried over from the previous code.

Reply:
I have not checked, but would not be surprised if they were not.  Our tests pay too little attention to `inet_res`.